app/models/mailing_list/signup.rb

```diff
@@ -0,0 +1,195 @@
+require "attribute_filter"
+
+module MailingList
+  class Signup
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+    include ActiveModel::Validations::Callbacks
+    include ::Wizard::ApiClientSupport
+
+    attribute :accept_privacy_policy, :string
+    attribute :accepted_policy_id, :string
+    attribute :address_postcode, :string
+    attribute :email, :string
+    attribute :first_name, :string
+    attribute :last_name, :string
+    attribute :preferred_teaching_subject_id, :string
+
+    attribute :channel_id, :integer
+    attribute :consideration_journey_stage_id, :integer
+    attribute :degree_status_id, :integer
+
+    attribute :timed_one_time_password
+
+    attribute :candidate_id
+    attribute :qualification_id
+    attribute :already_subscribed_to_mailing_list, :boolean
+
+    before_validation :strip_whitespace
+
+    validates :address_postcode,
+              postcode: true
+
+    validates :email,
+              presence: true,
+              email_format: true
+
+    validates :first_name,
+              presence: true,
+              length: { maximum: 256 }
+
+    validates :last_name,
+              presence: true,
+              length: { maximum: 256 }
+
+    validates :channel_id,
+              inclusion: { in: :channel_ids, allow_blank: true }
+
+    validates :consideration_journey_stage_id,
+              presence: true,
+              inclusion: { in: :consideration_journey_stage_ids }
+
+    validates :preferred_teaching_subject_id,
+              presence: true,
+              inclusion: { in: :teaching_subject_ids }
+
+    validates :degree_status_id,
+              presence: true,
+              inclusion: { in: :degree_status_option_ids }
+
+    validates :accept_privacy_policy,
+              acceptance: true,
+              allow_nil: false
+
+    validates :timed_one_time_password,
+              on: :verify,
+              presence: true,
+              length: { is: 6, message: :invalid },
+              format: { with: /\A[0-9]*\z/, message: :invalid }
+
+    validate :timed_one_time_password_is_correct,
+             on: :verify
+
+    def degree_status_options
+      @degree_status_options ||= query_degree_status
+    end
+
+    def teaching_subjects
+      @teaching_subjects ||= [OpenStruct.new(id: nil, value: "Please select")] + query_teaching_subjects
```

Comment:
Not a big deal, but could avoid having to do this by using `prompt: { "Please select" }` in the view I think